include/ensmallen_bits/scd/scd_impl.hpp

```diff
@@ -89,8 +91,12 @@ double SCD<DescentPolicyType>::Optimize(ResolvableFunctionType& function,
   Info << "SCD: maximum iterations (" << maxIterations << ") reached; "
       << "terminating optimization." << std::endl;
 
-  // Calculate and return final objective.
-  return function.Evaluate(iterate);
+  // Calculate and final objective if exactObjective is set to true.
+  if (exactObjective)
+  {
+    lastObjective = function.Evaluate(iterate);
+  }
+  return lastObjective;
```

Comment:
Just like CNE, SCD doesn't work on decomposable functions, so I think we can leave SCD as it is. :+1: